docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,578 @@
+(npe2-manifest-spec)=
+
+# npe2 manifest specification
+
+The **plugin manifest** is a specially formatted text file declaring the
+functionality of a [npe2][] plugin. A **plugin** is a python package that
+contains the manifest together with a suitable _[entry point group][epg]_ in
+the package metadata.
+
+Manifest files may be [json][], [yaml][], or [toml][] files conforming to the
+manifest schema. The **schema** defines what to expect in a manifest by
+defining the fields and their data types. These fields and their meanings are
+described below.
+
+A **plugin engine** is used to discover plugins, provide utilities for
+querying and manipulating plugins, and for exposing plugin-backed
+functionality to _napari_. **Discovery** is the process by that finds plugins,
+parses the manifests and indexes them for later use. The [npe2][] library
+manages these responsibilities.
+
+## Compatibility
```

Comment:
I don't know if this should be first up... Since it's a second level header and will be visible in the right side nav, I would suggest moving this section down to below the schema maybe even?

Reply:
moved down to the command section and added an internal link to the engine field.